src/sparseml/pytorch/sparsification/quantization/modifier_quantization.py

```diff
@@ -494,38 +551,105 @@ def _enable_module_qat(self, module: Module):
-            self._model_fuse_fn_kwargs["inplace"] = True
-            fuse_module_conv_bn_relus(module, **self._model_fuse_fn_kwargs)
 
-        activation_qconfig_kwargs = self._get_updated_activation_qconfig_kwargs()
+        # build list of layer types that should not quantize output activations
+        to_remove_layer_name = []
+        if not self._quantize_linear_activations:
+            to_remove_layer_name.extend(["Linear", "LinearReLU"])
+
+        if not self._quantize_conv_activations:
+            to_remove_layer_name.extend(
+                [
+                    "Conv1d",
+                    "Conv2d",
+                    "Conv3d",
+                    "ConvBn1d",
+                    "ConvBn2d",
+                    "ConvBn3d",
+                    "ConvReLU1d",
+                    "ConvReLU2d",
+                    "ConvReLU3d",
+                    "ConvBnReLU1d",
+                    "ConvBnReLU2d",
+                    "ConvBnReLU3d",
+                ]
+            )
+        if len(to_remove_layer_name) == 0:
+            to_remove_layer_name = None
+
+        # fix for freezing batchnorm statistics when not fusing BN with convs.
+        # pytorch only supports freezing batchnorm statistics for fused modules.
+        # this fix wraps BN modules adding with a new module class that supports
+        # methods related to freezing/unfreezing BN statistics.
+        configure_module_bn_wrappers(module)
+
+        # set qconfig.
+        # if tensorrt flag is used, set activation and weights to symmetric
+        # quantization.
+        # otherwise, use the default values set in get_qat_qconfig
+        if self.tensorrt:
+            _symmetric_activations = True
+            _activation_dtype = torch.qint8
+            _symmetric_weights = True
+            _weight_dtype = torch.qint8
+        else:
+            _symmetric_activations = None
+            _activation_dtype = None
+            _symmetric_weights = None
+            _weight_dtype = None
 
-        # prepare each module / submodule for quantization
         qconfig = get_qat_qconfig(
+            symmetric_activations=_symmetric_activations,
+            symmetric_weights=_symmetric_weights,
             reduce_range=self._reduce_range,
-            activation_qconfig_kwargs=activation_qconfig_kwargs,
+            activation_qconfig_kwargs=self.activation_qconfig_kwargs,
             weight_qconfig_kwargs=self.weight_qconfig_kwargs,
+            activation_dtype=_activation_dtype,
+            weight_dtype=_weight_dtype,
+            activation_bits=self.activation_bits,
+            weight_bits=self.weight_bits,
         )
+
+        # prepare each module / submodule for quantization
         for name, quant_module in self._modules_to_quantize:
             # wrap any modules with wrap_qat set to True as QATWrapper(s)
             configure_module_qat_wrappers(
                 quant_module,
+                symmetric_activations=_symmetric_activations,
```

Comment:
looking at all these repeated arguments, it may be time to pull some out into a dict or dataclass to make passing easier

Reply:
See reply above about new dataclass.